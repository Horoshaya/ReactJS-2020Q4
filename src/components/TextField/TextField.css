.label {
  color: #f65261;
  text-transform: uppercase;
}

.input {
  width: 100%;
  padding: 10px 0;
  margin: 10px 0;
  text-indent: 10px;
  color: #ffffff;
  background: #424242;
  border: none;
  border-radius: 4px;
}
