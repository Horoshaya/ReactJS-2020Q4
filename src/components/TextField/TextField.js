import React from 'react';
import styles from './TextField.css';

const TextField = ({ label, title, type }) => {
  return (
    <label className={styles.label}>
      {label}
      <input
        className={styles.input}
        type={type ? type : 'text'}
        defaultValue={title}
      />
    </label>
  );
};

export default TextField;
