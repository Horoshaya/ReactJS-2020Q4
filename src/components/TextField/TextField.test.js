import React from 'react';
import rerender from 'react-test-renderer';
import TextField from './TextField';
import { Formik } from 'formik';

describe('when TextField', () => {
  test('then snapshot created', () => {
    const component = rerender.create(
      <Formik>
        <TextField />
      </Formik>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
