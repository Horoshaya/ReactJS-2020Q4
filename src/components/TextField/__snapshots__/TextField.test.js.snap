// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when TextField then snapshot created 1`] = `
<label>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</label>
`;
